sql/api/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -433,7 +433,8 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
     stringConcat.append("struct<")
     var i = 0
     while (i < len) {
-      stringConcat.append(s"${fields(i).name}:${fields(i).dataType.catalogString}")
+      val name = QuotingUtils.quoteIfNeeded(fields(i).name)
+      stringConcat.append(s"$name:${fields(i).dataType.catalogString}")
```

Comment:
cc @zsxwing, this seems also affect output schema of queries

Reply:
Also, cc @cloud-fan. If this is the right option to go with, shall we create a variant for the `catalogString` method to minimize the impact?

Reply:
FYI, the previous discussion thread can be visited at https://github.com/apache/spark/pull/45039#discussion_r1857782815